gatsby-plugin-sharp 2.0.24 requires gatsby@2.1.20 but specifies peer dependency of gatsby@^2.0.0

## Description

gatsby-plugin-sharp 2.0.24 requires gatsby@2.1.20 but specifies peer dependency of gatsby@^2.0.0

https://github.com/gatsbyjs/gatsby/pull/10257
https://github.com/gatsbyjs/gatsby/blob/8e1661806931b0af42b744b9b5c8cf2909ae1201/packages/gatsby-plugin-sharp/package.json#L43

### Steps to reproduce

Use the right combination of plugins and configs to actuate gatsby-plugin-sharp and then at build time:

```
error Plugin gatsby-plugin-sharp returned an error


  Error: Cannot find module 'gatsby/dist/utils/cpu-core-count'
  
  - v8-compile-cache.js:159 require
    [code]/[v8-compile-cache]/v8-compile-cache.js:159:20
  
  - index.js:88 Object.<anonymous>
    [code]/[gatsby-plugin-sharp]/index.js:88:22
  
  - v8-compile-cache.js:178 Module._compile
    [code]/[v8-compile-cache]/v8-compile-cache.js:178:30
  
  - v8-compile-cache.js:159 require
    [code]/[v8-compile-cache]/v8-compile-cache.js:159:20
  
  - gatsby-node.js:3 Object.<anonymous>
    [code]/[gatsby-plugin-sharp]/gatsby-node.js:3:18
  
  - v8-compile-cache.js:178 Module._compile
    [code]/[v8-compile-cache]/v8-compile-cache.js:178:30
```

### Expected result

Minor version updates to not break the install.

### Actual result

Breakage.

--- a/packages/gatsby-plugin-manifest/src/gatsby-node.js
+++ b/packages/gatsby-plugin-manifest/src/gatsby-node.js
@@ -6,11 +6,16 @@ const { defaultIcons, doesIconExist } = require(`./common.js`)
 
 sharp.simd(true)
 
-// Handle Sharp's concurrency based on the Gatsby CPU count
-// See: http://sharp.pixelplumbing.com/en/stable/api-utility/#concurrency
-// See: https://www.gatsbyjs.org/docs/multi-core-builds/
-const cpuCoreCount = require(`gatsby/dist/utils/cpu-core-count`)
-sharp.concurrency(cpuCoreCount())
+try {
+  // Handle Sharp's concurrency based on the Gatsby CPU count
+  // See: http://sharp.pixelplumbing.com/en/stable/api-utility/#concurrency
+  // See: https://www.gatsbyjs.org/docs/multi-core-builds/
+  const cpuCoreCount = require(`gatsby/dist/utils/cpu-core-count`)
+  sharp.concurrency(cpuCoreCount())
+} catch {
+  // if above throws error this probably means that used Gatsby version
+  // doesn't support cpu-core-count utility.
+}
 
 function generateIcons(icons, srcIcon) {
   return Promise.map(icons, icon => {

--- a/packages/gatsby-plugin-sharp/src/index.js
+++ b/packages/gatsby-plugin-sharp/src/index.js
@@ -62,11 +62,16 @@ Promise.promisifyAll(sharp.prototype, { multiArgs: true })
 // adventurous and see what happens with it on.
 sharp.simd(true)
 
-// Handle Sharp's concurrency based on the Gatsby CPU count
-// See: http://sharp.pixelplumbing.com/en/stable/api-utility/#concurrency
-// See: https://www.gatsbyjs.org/docs/multi-core-builds/
-const cpuCoreCount = require(`gatsby/dist/utils/cpu-core-count`)
-sharp.concurrency(cpuCoreCount())
+try {
+  // Handle Sharp's concurrency based on the Gatsby CPU count
+  // See: http://sharp.pixelplumbing.com/en/stable/api-utility/#concurrency
+  // See: https://www.gatsbyjs.org/docs/multi-core-builds/
+  const cpuCoreCount = require(`gatsby/dist/utils/cpu-core-count`)
+  sharp.concurrency(cpuCoreCount())
+} catch {
+  // if above throws error this probably means that used Gatsby version
+  // doesn't support cpu-core-count utility.
+}
 
 const bar = new ProgressBar(
   `Generating image thumbnails [:bar] :current/:total :elapsed secs :percent`,

